ikarus/finiteelements/mechanics/kirchhoffloveshell.hh

```diff
@@ -180,30 +203,179 @@ namespace Ikarus {
      * @param local The local coordinates at which results are to be calculated.
      * @param result The result container to store the calculated values.
      */
-    void calculateAt([[maybe_unused]] const ResultRequirementsType& req,
-                     [[maybe_unused]] const Dune::FieldVector<double, Traits::mydim>& local,
-                     [[maybe_unused]] ResultTypeMap<double>& result) const {
+    void calculateAt([[maybe_unused]] const ResultRequirementsType &req,
+                     [[maybe_unused]] const Dune::FieldVector<double, Traits::mydim> &local,
+                     [[maybe_unused]] ResultTypeMap<double> &result) const {
       DUNE_THROW(Dune::NotImplemented, "No results are implemented");
     }
 
     Dune::CachedLocalBasis<
         std::remove_cvref_t<decltype(std::declval<LocalView>().tree().child(0).finiteElement().localBasis())>>
         localBasis;
-    std::function<Eigen::Vector<double, Traits::worlddim>(const Eigen::Vector<double, Traits::worlddim>&,
-                                                          const double&)>
+    std::function<Eigen::Vector<double, Traits::worlddim>(const Dune::FieldVector<double, Traits::worlddim> &,
+                                                          const double &)>
         volumeLoad;
-    std::function<Eigen::Vector<double, Traits::worlddim>(const Eigen::Vector<double, Traits::worlddim>&,
-                                                          const double&)>
+    std::function<Eigen::Vector<double, Traits::worlddim>(const Dune::FieldVector<double, Traits::worlddim> &,
+                                                          const double &)>
         neumannBoundaryLoad;
-    const BoundaryPatch<GridView>* neumannBoundary;
+    const BoundaryPatch<GridView> *neumannBoundary;
     mutable Dune::BlockVector<Dune::RealTuple<double, Traits::dimension>> dispAtNodes;
+    DefaultMembraneStrain membraneStrain;
     double emod_;
     double nu_;
     double thickness_;
     size_t numberOfNodes{0};
     int order{};
 
   protected:
+    auto computeMaterialAndStrains(const Dune::FieldVector<double, 2> &gpPos, int gpIndex, const Geometry &geo,
```

Comment:
add doxygen comment here and explain the return type? 

May be also add a short description in documentation about stress-resultant based implementation and explain the equations used or shall we do it in a different PR?